.circleci/config.yml

```diff
@@ -321,14 +331,111 @@ jobs:
+            /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+            eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+            brew install gh
+
+      - run:
+          name: Configure GitHub CLI Settings
+          command: echo 'export GH_REPO=$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME' >> $BASH_ENV
+
+      - run:
+          name: Look up the related GitHub PR number and branch name
+          command: |
+            eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+            echo "export GH_PR_NUMBER=$(gh api repos/${GH_REPO}/commits/${CIRCLE_SHA1}/pulls | jq '.[0] | .number')" >> $BASH_ENV
+            echo "export GH_PR_BRANCH=$(gh api repos/${GH_REPO}/commits/${CIRCLE_SHA1}/pulls | jq '.[0] | .head.ref')" >> $BASH_ENV
+
+      - run:
+          name: Ensure that version tag matches branch version
+          command: |
+            if [ "$(echo $GH_PR_BRANCH | tr -d "release/")" != "$CIRCLE_TAG" ]
+            then
+              echo "ERROR: Version number from tag does not match the version number from the branch name."
+              exit 1
+            fi
+
+      - configure-build-env
+
+      - run:
+          name: Set release version from tag name
+          command: echo 'export STREAMLIT_RELEASE_VERSION=$(echo $CIRCLE_TAG)' >> $BASH_ENV
+
+      - run:
+          name: Update version
+          command: python scripts/update_version.py $STREAMLIT_RELEASE_VERSION
+
+      - run:
+          name: Commit version updates
+          command: |
+            git config user.email "core+streamlitbot-github@streamlit.io"
+            git config user.name "Streamlit Bot"
+
+            git commit -am "Up version to $STREAMLIT_RELEASE_VERSION [skip ci]" && git push origin release/$CIRCLE_TAG || echo "No changes to commit"
+
+      # Password added to circleci environment
+      # https://ui.circleci.com/settings/project/github/streamlit/streamlit/environment-variables
+      - run:
+          name: init .pypirc
+          command: |
+            cd lib
+            echo -e "[pypi]" >> ~/.pypirc
+            echo -e "username = streamlit" >> ~/.pypirc
+            echo -e "password = $PYPI_PASSWORD" >> ~/.pypirc
+
+      - run:
+          name: Make package
+          no_output_timeout: 2h
+          command: |
+            echo "working directory = $(pwd)"
+            chmod +x ./frontend/scripts/preload-bokeh.sh
+            ${SUDO} apt-get install rsync
+            make package
+
+      - run:
+          name: Run CLI regression tests
+          command: make cli-regression-tests
+
+      - store_artifacts:
+          path: ./lib/dist
+
+      - run:
+          name: Upload to pypi
+          command: |
+            make distribute
+
+      - run:
+          name: Create GitHub Release and merge PR
+          command: |
+            eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+            gh release create $CIRCLE_TAG -n "" -t $CIRCLE_TAG
```

Comment:
Another side note, if it's ever desired to upload the release artifacts to the GitHub release, it's super easy to do so:

`gh release upload $CIRCLE_TAG dist/lib/*`